jubatus/core/anomaly/lof_storage.cpp

```diff
@@ -132,7 +137,21 @@ void lof_storage::get_all_row_ids(vector<string>& ids) const {
   nn_engine_->get_all_row_ids(ids);
 }
 
-void lof_storage::update_row(const string& row, const common::sfv_t& diff) {
+bool lof_storage::update_row(const string& row, const common::sfv_t& diff) {
+  if (ignore_kth_same_point_) {
+    std::vector<std::pair<std::string, float> > nn_result;
+
+    // Find k-1 NNs for the given sfv.
+    // If the distance to the (k-1) th neighbor is 0, the model already
+    // have (k-1) points that have the same feature vector as given sfv.
+    nn_engine_->neighbor_row(
+        diff, nn_result, neighbor_num_ - 1);
```

Comment:
I think we need to search neighbors by `row` ID instead of `diff` as `update_row` of recommender is a incremental update.